frontend/views/containers/payments/RecordPayment.vue

```diff
@@ -199,20 +200,36 @@ export default ({
             paymentType: PAYMENT_TYPE_MANUAL,
             ...(memo ? { memo } : {}) // TODO/BUG with flowTyper validation. Empty string '' fails.
           }
-          const msg = await sbp('gi.actions/group/payment', {
-            contractID: this.currentGroupId, data: paymentInfo
-          })
-          // TODO: hack until /payment supports sending completed payment
-          //       (and "uncompleting" a payment)
-          await sbp('gi.actions/group/paymentUpdate', {
-            contractID: this.currentGroupId,
-            data: {
-              paymentHash: msg.hash(),
-              updatedProperties: {
-                status: PAYMENT_COMPLETED
+
+          if (isStatusNotReceived) {
+            // If it's re-sending the payment that has been marked as 'not-recieved' by the receiver,
+            // only update the details of the existing payment item so that it doesn't lead to duplication bug in the payment UI.
+            await sbp('gi.actions/group/paymentUpdate', {
+              contractID: this.currentGroupId,
+              data: {
+                paymentHash: payment.hash,
+                updatedProperties: {
+                  ...paymentInfo,
+                  status: PAYMENT_COMPLETED
+                }
               }
-            }
-          })
+            })
+          } else {
+            const msg = await sbp('gi.actions/group/payment', {
+              contractID: this.currentGroupId, data: paymentInfo
+            })
+            // TODO: hack until /payment supports sending completed payment
+            //       (and "uncompleting" a payment)
+            await sbp('gi.actions/group/paymentUpdate', {
+              contractID: this.currentGroupId,
+              data: {
+                paymentHash: msg.hash(),
+                updatedProperties: {
+                  status: PAYMENT_COMPLETED
+                }
+              }
+            })
+          }
```

Comment:
What is happening here? AFAIK these changes were already included in `e2e-protocol` or `master`, is that right?

These changes, as well as others above, seem unrelated to the push API, and already integrated somewhere.. I think with some git magic and re-branching you should be able to figure out how to make these disappear from this PR (or a separate PR that's re-sent)?

Reply:
I think all these unrelated changes are from `master` branch and looks like it happened in [this commit ](https://github.com/okTurtles/group-income/pull/1770/commits/834299de8021ac2e9c05cb18208d7fb69a468053) when what I wanted/should have done was merging the latest `e2e-protocol` branch instead.